src/rc.js

```diff
@@ -0,0 +1,101 @@
+import { readFileSync } from 'fs';
+import rc from 'rc';
+
+import parse from './lockfile/parse.js';
+
+// Keys that will be propagated into the command line as arguments
+const ARG_KEYS = [
+    'production',
+
+    'offline',
+    'prefer-offline',
+    'proxy',
+    'https-proxy',
+    'network-concurrency',
+
+    'strict-semver',
+    'ignore-engines',
+    'ignore-optional',
+    'ignore-platform',
+    'ignore-scripts',
+    'skip-integrity-check',
+
+    'no-lockfile',
+    'pure-lockfile',
+    'frozen-lockfile',
+
+    'no-bin-links',
+    'link-duplicates',
+    'flat',
+
+    'cache-folder',
+    'modules-folder',
+    'global-folder',
+
+    'mutex',
+
+    'no-emoji',
+    'har',
```

Comment:
I've been wondering the same thing, but I'm a bit worried about the potential effects of letting the users put any argument into their rc files :/ For example, `--force` in the yarnrc seems like a bad idea.